### PR TITLE
Add initial level generation

### PR DESCRIPTION
**Pull request by kdhageman**
Adds the `LevelGenerator`, which generates a simple level at startup. 

**Note**: The map is not even close to being finished.

Example:
screen shot 2017-07-16 at 13 35 48

screen shot 2017-07-16 at 13 37 01

### CODE REVIEW

File: Assets/Animations.meta [hunk 1]
```diff
@@ -1,9 +0,0 @@
-fileFormatVersion: 2
```
Comment (ratenbuuren) [inline Assets/Animations.meta]: Unnecessary change

File: Assets/Materials.meta [hunk 2]
```diff
@@ -1,9 +0,0 @@
-fileFormatVersion: 2
```
Comment (ratenbuuren) [inline Assets/Materials.meta]: Unnecessary change

File: Assets/Scripts/LevelGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	private void generateEdges() {
+		int[] arr = { 1, -1 };
+
+		// create top and bottom row
+		for (int x = 0; x < width+2; x++) { 
+			float xPos = (float)x - (width / 2f) - 0.5f;
+			float yPos = (height / 2f) + 0.5f;
+
+			foreach (int i in arr) {
```
Comment (ratenbuuren) [inline Assets/Scripts/LevelGenerator.cs]: Foreach can be abstracted into method. Emphasis on could ;)

File: Assets/Scripts/LevelGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		private GameObject template;
+
+		public PrefabBuilder(GameObject template) {
+			this.template = template;
+			this.obj = Instantiate(template);
+		}
+
+		public PrefabBuilder position(float x, float y) {
+			obj.transform.position = new Vector2 (x, y);
+			obj.name = String.Format ("{0} [{1},{2}]", template.name, x, y);
```
Comment (ratenbuuren) [inline Assets/Scripts/LevelGenerator.cs:100]: Nice!, will definitely start using this naming as well 👍

File: Assets/Sounds.meta [hunk 5]
```diff
@@ -1,9 +0,0 @@
-fileFormatVersion: 2
```
Comment (ratenbuuren) [inline Assets/Sounds.meta]: Unnecessary change

File: Assets/Sprites/topdowntanks/PNG/Environment/treeLarge.png.meta [hunk 6]
```diff
@@ -40,7 +40,7 @@ TextureImporter:
   alignment: 0
   spritePivot: {x: 0.5, y: 0.5}
   spriteBorder: {x: 0, y: 0, z: 0, w: 0}
-  spritePixelsToUnits: 100
+  spritePixelsToUnits: 107
```
Comment (ratenbuuren) [inline Assets/Sprites/topdowntanks/PNG/Environment/treeLarge.png.meta]: Would recommend also using 128. Consistent and always fits nicely on cubes below and tiles

File: Assets/Sprites/topdowntanks/PNG/Environment/treeSmall.png.meta [hunk 7]
```diff
@@ -40,7 +40,7 @@ TextureImporter:
   alignment: 0
   spritePivot: {x: 0.5, y: 0.5}
   spriteBorder: {x: 0, y: 0, z: 0, w: 0}
-  spritePixelsToUnits: 100
+  spritePixelsToUnits: 87
```
Comment (ratenbuuren) [inline Assets/Sprites/topdowntanks/PNG/Environment/treeSmall.png.meta]: Same here

**Review by ratenbuuren — CHANGES_REQUESTED** [review]
Some small comments. Looks good for the MVP :) +1

File: Assets/Animations.meta (hunk 1 above)
Comment (kdhageman) [reply Assets/Animations.meta]: Unity refuses me to keeps these files 😞 According to the log messages, the meta files cannot exist with the associated asset.

**Comment by kdhageman** [comment]
@ratenbuuren Can you review again?

File: Assets/Animations.meta (hunk 1 above)
Comment (ratenbuuren) [reply Assets/Animations.meta]: Hmm weird. We can look into that. You could use git checkout origin/master -- file path to check them out and push the originals. This will exclude them from this PR until we fix it

**Review by ratenbuuren — APPROVED** [review]
